mkosi/config.py

```diff
@@ -5146,6 +5156,15 @@ def parse_config(
     # One of the specifiers needs access to the directory, so make sure it is available.
     context.config["directory"] = args.directory
 
+    # Config paths need to be absolute - as a special case, if it is not, look in the parent
+    # of --directory. This works nicely on OBS, where the --directory is the git repository
+    # and the --default is the mkosi.conf selected at source service time, placed in the parent.
+    if args.default_config is not None:
+        if args.directory is not None and not args.default_config.is_absolute():
+            context.default_config = args.directory.parent / args.default_config
+        else:
+            context.default_config = args.default_config
```

Comment:
Since the `args.default_config` is `None` anyways, this could be shortened into
```suggestion
    if (
        args.default_config is not None:
        and args.directory is not None 
        and not args.default_config.is_absolute()
    ):
        context.default_config = args.directory.parent / args.default_config
```
(well, it's longer, but slightly less complex)

Reply:
With that change, unless args.directory is set, args.default_config is ignored, even if it's an absolute path which would otherwise work